sources/parsers/parsers.ml

```diff
@@ -132,10 +158,25 @@ let parse_input_file file =
   in
   try
     (if Options.infer_output_format () then
-       match ext with
-       | ".smt2" | ".psmt2" -> Options.set_output_format OSmtlib
-       | ".why" | ".mlw" | ".ae" -> Options.set_output_format ONative
-       | _ -> Format.eprintf "Warning: This extension is not supported@."
+       let ext =
+         try Str.string_after ext 1
```

Comment:
The `Str` module is almost deprecated(slow, outdated, barely maintained, etc...) and absolutely not warranted in this case. Here, you can use `Filename.extension` (since ocaml >= 4.04), or simply use `String.rindex_from` followed by a `String.sub` to have greater compatibility across ocaml versions.